### PR TITLE
Add login and signup

### PR DESCRIPTION
**Pull request by Michael-R-Mac**

### CODE REVIEW

File: src/components/LoginView/LoginView.jsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    fetch("https://cf-movie-api.herokuapp.com/login", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify(data)
+    })
+    .then((response) => response.json())
+    .then((data) => {
+      console.log("Login response: ", data);
```
Comment (frankhn) [inline src/components/LoginView/LoginView.jsx:25]: Always remove console logs when pushing or merging to the main branch

File: src/components/LoginView/LoginView.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      access: username,
+      secret: password
+    };
+
+    fetch("https://cf-movie-api.herokuapp.com/login", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify(data)
```
Comment (frankhn) [inline src/components/LoginView/LoginView.jsx:21]: You should be sending json not strings.

Just like this
{
access: username,
  secret: password
}

File: src/components/SignUpView/SignUpView.jsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    const data = {
+      Username: username,
+      Password: password,
+      Email: email,
+      Birthday: birthday
+    };
+
+    fetch("https://cf-movie-api.herokuapp.com/users", {
+      method: "POST",
+      body: JSON.stringify(data),
```
Comment (frankhn) [inline src/components/SignUpView/SignUpView.jsx:21]: Same as the above

File: src/components/MainView/MainView.jsx [hunk 4]
```diff
@@ -1,11 +1,22 @@
 import { useState, useEffect } from "react";
 import { MovieCard } from "../MovieCard/MovieCard";
 import { MovieView } from "../MovieView/MovieView";
+import { LoginView } from "../LoginView/LoginView";
+import { SignupView } from "../SignUpView/SignUpView";
 
 export const MainView = () => {
+  const storedUser = JSON.parse(localStorage.getItem("user"));
+  const storedToken = localStorage.getItem("token");
+  const [user, setUser] = useState(storedUser? storedUser : null);
```
Comment (frankhn) [inline src/components/MainView/MainView.jsx:10]: Just the default value to null as your not using type.

**Review by frankhn — CHANGES_REQUESTED** [review]
Kindly review the comments above

File: src/components/LoginView/LoginView.jsx (hunk 2 above)
Comment (Michael-R-Mac) [reply src/components/LoginView/LoginView.jsx:21]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify

It is a json string, not a string. The exercise content also recommends using json.stringify.

File: src/components/LoginView/LoginView.jsx (hunk 1 above)
Comment (Michael-R-Mac) [reply src/components/LoginView/LoginView.jsx:25]: Thank you removed.

File: src/components/MainView/MainView.jsx (hunk 4 above)
Comment (Michael-R-Mac) [reply src/components/MainView/MainView.jsx:10]: I do not understand this one.

File: src/components/LoginView/LoginView.jsx (hunk 2 above)
Comment (Michael-R-Mac) [reply src/components/LoginView/LoginView.jsx:21]: I changed it to:     
const data = {
      Username: username,
      Password: password
    };

And it is working well. Thank you for the tip.

**Pull request merged**